File: Philosophers_sharp/Philosophers_sharp/Philosopher.cs
using System;
using System.Threading;

namespace Philosophers_sharp
{
    public class Philosopher
    {
        private int _id;
        private int _leftForkId;
        private int _rightForkId;
        private Table _table;
        private Personal _personal;
        private int _lapsCount;
        private int _thinkingTime_ms;
        private int _eatingTime_ms;

        public Philosopher(int id, int leftForkId, int rightForkId, Table table, Personal personal, int lapsCount, int thinkingTime_ms, int eatingTime_ms)
        {
            _id = id;
            _leftForkId = leftForkId;
            _rightForkId = rightForkId;
            _table = table;
            _personal = personal;
            _lapsCount = lapsCount;
            _thinkingTime_ms = thinkingTime_ms;
            _eatingTime_ms = eatingTime_ms;
        }

        public void Start()
        {
            for (int i = 0; i < _lapsCount; i++)
            {
                //Console.WriteLine($"Philosopher {_id} thinking");
                Thread.Sleep(_thinkingTime_ms);

                while (true)
                {
                    bool canStop = _personal.CallWaiter(_id);
                    if (canStop)
                    {
                        break;
                    }
                }
                _table.WaitOneFork(_leftForkId);
                _table.WaitOneFork(_rightForkId);

                //Console.WriteLine($"Philosopher {_id} eating");
                Thread.Sleep(_eatingTime_ms);
                Counter.Counters[_id]++;

                _personal.RecallWaiter(_id);
                _table.ReleaseFork(_leftForkId);
                _table.ReleaseFork(_rightForkId);
            }
        }
    }
}

File: Philosophers_java/src/Philosopher.java
public class Philosopher extends Thread{
    private Integer _id;
    private Integer _leftForkId;
    private Integer _rightForkId;
    private Table _table;
    private Manager _manager;
    private Integer _lapsCount;
    private Integer _thinkingTime;
    private Integer _eatingTime;

    public Philosopher(Integer id,
                       Integer leftForkId,
                       Integer rightForkId,
                       Table table,
                       Manager manager,
                       Integer lapsCount,
                       Integer thinkingTime,
                       Integer eatingTime){
        _id = id;
        _leftForkId = leftForkId;
        _rightForkId = rightForkId;
        _table = table;
        _manager = manager;
        _lapsCount = lapsCount;
        _thinkingTime = thinkingTime;
        _eatingTime = eatingTime;
    }

    @Override
    public void run(){
        for (Integer i = 0; i < _lapsCount; i++){
            try {
                //System.out.println("Philosopher " + _id + " thinking");
                Thread.sleep(_thinkingTime);

                while(true){
                    Boolean canEating = _manager.canEating(_id);
                    if(canEating){
                        break;
                    }
                }
                _table.acquireFork(_leftForkId);
                _table.acquireFork(_rightForkId);

                //System.out.println("Philosopher " + _id + " eating");
                Thread.sleep(_eatingTime);
                Main.Counter[_id]++;

                _manager.stopEating(_id);
                _table.releaseFork(_leftForkId);
                _table.releaseFork(_rightForkId);
            } catch (InterruptedException e) {
                throw new RuntimeException(e);
            }
        }
    }
}